### PR TITLE
Spelling driver

### PR DESCRIPTION
**Pull request by jsoref**
This fixes some misspellings in Swift driver*, it is split per https://github.com/apple/swift/pull/42421#issuecomment-1101092698

### CODE REVIEW

**Comment by tshortli** [comment]
@swift-ci please test

**Comment by tshortli** [comment]
@swift-ci please test Windows

File: lib/Driver/Compilation.cpp [hunk 1]
```diff
@@ -1705,7 +1705,7 @@ static void writeIndexUnitOutputPathsToFilelist(llvm::raw_fd_ostream &out,
   for (auto &output : outputInfo.getIndexUnitOutputFilenames())
     out << output << "\n";
 }
-static void writeSupplementarOutputToFilelist(llvm::raw_fd_ostream &out,
+static void writeSupplementaryOutputToFilelist(llvm::raw_fd_ostream &out,
                                               const Job *job) {
```
Comment (CodaFi) [inline lib/Driver/Compilation.cpp:1709]: Nit: Please re-run clang format to re-align this

**Review by CodaFi — APPROVED** [review]

File: lib/Driver/Compilation.cpp (hunk 1 above)
Comment (jsoref) [reply lib/Driver/Compilation.cpp:1709]: A naive run of `clang-format version 14.0.1` (from `brew install clang-format`) results in this file being dramatically rewritten.

I'm happy to fix the indentation of the line below, but I'm uncomfortable rewriting the entire file (or risking rewriting the entire repository).

**Comment by tshortli** [comment]
@swift-ci please smoke test

File: lib/Driver/Compilation.cpp (hunk 1 above)
Comment (tshortli) [reply lib/Driver/Compilation.cpp]: The recommended workflow is to use `git-clang-format <branch base SHA>` to just reformat the lines touched on the branch. A bit more documentation here: https://github.com/apple/swift/blob/main/docs/HowToGuides/FAQ.md#how-do-i-format-my-changes

**Pull request merged**